accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/DatabaseService.java

```diff
@@ -30,79 +30,127 @@
 import java.util.List;
 
 /**
- * Interface to the database service that handles the storage and queries of a object with their hashed version and
+ * Interface to the database service that handles the storage and queries of an object with their hashed version and
  * accession.
  *
- * @param <MODEL>
- * @param <HASH>
- * @param <ACCESSION>
+ * @param <MODEL> Type of the objects identified by the accessions
+ * @param <HASH> Type of the hash calculated based on the fields that uniquely identify an accessioned object
+ * @param <ACCESSION> Accession of the object that identify a particular model
  */
 public interface DatabaseService<MODEL, HASH, ACCESSION> {
 
     /**
      * Finds all valid accessioned model data that has a hashed message in the collection @param hashes.
      *
-     * @param hashes
-     * @return
+     * @param hashes Hash value of objects
+     * @return Wrappers containing the objects and their associated accession and hash
```

Comment:
A list!

Reply:
List and collection are java specific. Should we include them? Would plural Wrappers be enough?

Reply:
I didn't mean the `List` in Java, I meant "list" in plain english. I find `a list of wrapper objects` more specific in that it returns a `List` of wrapper objects while with `Wrappers` I will always have to check the return type. Maybe it is just me so if you are happy with plural wrappers then please ignore it.